apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/PatientResourceProvider.java

```diff
@@ -350,36 +353,96 @@ private CcwCodebookVariable partDCwVariableFor(String system) {
    * @param field to match on
    * @param value to match on
    * @param paging to use for the result set
-   * @param joins to add for many-to-one relations
+   * @param identifiers to add for many-to-one relations
    * @return the criteria
    */
-  private CriteriaQuery<Beneficiary> queryBeneficiariesBy(
-      SingularAttribute<Beneficiary, String> field,
-      String value,
-      PatientLinkBuilder paging,
-      List<SetAttribute<Beneficiary, ?>> joins) {
+  private TypedQuery<Beneficiary> queryBeneficiariesBy(
+      String field, String value, PatientLinkBuilder paging, List<String> identifiers) {
+    String joinsClause = "";
+    if (hasMBI(identifiers)) joinsClause += "left join fetch b.medicareBeneficiaryIdHistories ";
+    if (hasHICN(identifiers)) joinsClause += "left join fetch b.beneficiaryHistories ";
 
-    CriteriaBuilder cb = entityManager.getCriteriaBuilder();
-    CriteriaQuery<Beneficiary> criteria = cb.createQuery(Beneficiary.class);
-    Root<Beneficiary> b = criteria.from(Beneficiary.class);
-    joins.forEach(f -> b.fetch(f, JoinType.LEFT));
+    if (paging.isPagingRequested() && !paging.isFirstPage()) {
+      String query =
+          "select b from Beneficiary b "
+              + joinsClause
+              + "where b."
+              + field
+              + " = :value and b.beneficiaryId > :cursor "
+              + "order by b.beneficiaryId asc";
+
+      return entityManager
+          .createQuery(query, Beneficiary.class)
+          .setParameter("value", value)
+          .setParameter("cursor", paging.getCursor());
+    } else {
+      String query =
+          "select b from Beneficiary b "
+              + joinsClause
+              + "where b."
+              + field
+              + " = :value "
+              + "order by b.beneficiaryId asc";
+
+      return entityManager.createQuery(query, Beneficiary.class).setParameter("value", value);
+    }
+  }
 
```

Comment:
Switching over to the JPQL style for these queries. Easier to understand and read to my eyes. Simpler code as well.

Reply:
I like this and is easier to read for me too!